framework/db/mssql/Schema.php

```diff
@@ -126,6 +126,24 @@ protected function resolveTableName($name)
         return $resolvedName;
     }
 
+    /**
+     * {@inheritDoc}
+     * @param $name
+     * @return array|mixed
+     */
+    protected function getTableNameParts($name)
+    {
+        $parts = [$name];
+        preg_match_all('/([^.\[\]]+)|\[([^\[\]]+)\]/', $name, $matches);
+        if (isset($matches[0]) && is_array($matches[0]) && !empty($matches[0])) {
+            $parts = $matches[0];
+        }
+
+        $parts = str_replace(['[', ']'], '', $parts);
```

Comment:
If the `if` above we assume that `$parts` is an array. Then we're doing `str_replace` on it. That would cause error.

Reply:
Why? `str_replace` works with arrays.

Reply:
You're right. PHP standard library is a mess :)